sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -327,7 +327,7 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
    */
   def replace[T](col: String, replacement: Map[T, T]): DataFrame = {
     if (col == "*") {
-      replace0(df.columns, replacement)
+      replace0(df.columns.map { colName => s"`${colName}`" }, replacement)
```

Comment:
it may not work if `colName` already includes backticks. I think we should let `replace0` take `Seq[Attribute]` and here we pass in `df.logicalPlan.output`

Reply:
makes sense.